sdk/identity/identity/src/credentials/managedIdentityCredential/imdsMsi.ts

```diff
@@ -188,12 +197,13 @@ export const imdsMsi: MSI = {
     let nextDelayInMs = imdsMsiRetryConfig.startDelayInMs;
     for (let retries = 0; retries < imdsMsiRetryConfig.maxRetries; retries++) {
       try {
-        return await msiGenericGetToken(
-          identityClient,
-          prepareRequestOptions(scopes, clientId),
-          expiresInParser,
-          getTokenOptions
-        );
+        const request = createPipelineRequest({
+          abortSignal: getTokenOptions.abortSignal,
+          ...prepareRequestOptions(scopes, clientId),
+          allowInsecureConnection: true
```

Comment:
I think we ought to document the exact reasoning for any settings like this where we disable an otherwise default security setting.

Reply:
Most of the IMDS endpoints are `http` endpoints. The default one is `http://169.254.169.254/`, but environment variables can specify another endpoint. I will add a comment mentioning this.